app/workers/travis_branch_monitor.rb

```diff
@@ -0,0 +1,95 @@
+require 'travis'
+
+class TravisBranchMonitor
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot_glacial, :retry => false
+
+  include Sidetiq::Schedulable
+  recurrence { hourly.minute_of_hour(0, 15, 30, 45) }
+
+  include SidekiqWorkerMixin
+
+  class << self
+    private
+
+    # For this class, repos will sometimes the repo needs to be mapped to a
+    # specific gitter room, so a hash is required.
+    #
+    # This override allows for doing this in the config
+    #
+    #   travis_branch_monitor:
+    #     included_repos:
+    #       ManageIQ/manageiq-ui-classic: ManageIQ/ui
+    #       ManageIQ/manageiq-gems-pending: ManageIQ/core
+    #       ManageIQ/manageiq:
+    #       ManageIQ/miq_bot:
+    #
+    # Which you are allowed to leave the value empty, and the key will be used
+    # where appropriate (not used in this class).
+    #
+    # The result from the above for this method will then be:
+    #
+    #   [
+    #     "ManageIQ/manageiq-ui-classic",
+    #     "ManageIQ/manageiq-gems-pending",
+    #     "ManageIQ/manageiq",
+    #     "ManageIQ/miq_bot"
+    #   ]
+    #
+    def included_and_excluded_repos
+      super # just used for error handling...
+
+      [
+        settings.included_repos.try(:to_h).try(:stringify_keys).try(:keys),
+        settings.excluded_repos.try(:to_h).try(:stringify_keys).try(:keys)
+      ]
+    end
+  end
+
+  def perform
+    if !first_unique_worker?
+      logger.info("#{self.class} is already running, skipping")
+    else
+      process_repos
+    end
+  end
+
+  def process_repos
+    enabled_repos.each do |repo|
+      process_repo(repo)
+    end
+  end
+
+  def process_repo(repo)
+    # repo.regular_branch_names.each do |branch|
```

Comment:
Most definitely.  This code definitely is subject to "copy-pasta syndrome".

Reply:
Followup:  I didn't review this comment without looking at the context.

So I think this is related to the `TODO` that is below:

```ruby
# TODO:  Handle this for all release branches
process_branch(repo, 'master')
```

An probably what should be done is we should be doing `repo.regular_branches` (?) and passing the `branch_record` directly into `process_branch` instead of doing the lookup there.  That said, unsure if `regular_branches` does what we want in this case.  Would have to look.

That is also a question if we want to first start with 'master' for a first pass to test this feature out and then move on to the release branches going forward.  That way we can determine if this is indeed the route we want to take with this, or if it is causing too much noise in chat.

Reply:
We probably don't want to track all of the `regular_branches` it probably only makes sense for a subset (master, jansa, ivanchuk, hammer) or something like that.  Maybe that belongs in a setting?

Reply:
Yeah, that probably makes sense.

This probably falls in the "doesn't need to be a private setting" category argument that I have been making, but we can adjust that at a later date.  I will look into making this a regular setting for now.  👍

Reply:
> We probably don't want to track all of the regular_branches it probably only makes sense for a subset (master, jansa, ivanchuk, hammer) or something like that. Maybe that belongs in a setting?

We only track branches we care about, so I think all regular branches is ok.

Reply:
We might want to do some cleanup then because we are tracking everything back to darga.

Reply:
Currently using `.regular_branches` for this, so I think that has resolved this from a code aspect.  Leaving this unresolved just so it isn't lost as something we might want to look at on prod.